src/main/java/nl/tudelft/contextproject/model/Game.java

```diff
@@ -89,4 +104,35 @@ public Level getLevel() {
 	public void setLevel(Level level) {
 		this.level = level;
 	}
+	
+	/**
+	 * End this game.
+	 * 
+	 * @param didElvesWin
+	 * 		true when the elves won, false when the dwarfs did
+	 */
+	public void endGame(boolean didElvesWin) {
+		controller.gameEnded(didElvesWin);
+	}
+
+	/**
+	 * Update the game timer.
+	 * 
+	 * @param tpf
+	 * 		the time per frame of this update
+	 */
+	public void update(float tpf) {
+		timeLimit -= tpf;
```

Comment:
The tpf value changes depending on the framerate. A higher framerate will make the tpf smaller and vice versa. In this way adding tpf until it reaches 1 takes exactly one second. Therefore we can use the time per frame as a clock substitute.